### PR TITLE
Implementing support for secrets

### PR DESCRIPTION
**Pull request by schrieveslaach**
Add configuration to define secrets for specific services for specific apps. These
secrets will be mounted in the container under `/run/secrets/<name>` so that the
service is able to use e.g. database credentials.

### CODE REVIEW

File: api/src/models/service.rs [hunk 1]
```diff
@@ -69,6 +69,7 @@ pub struct ServiceConfig {
     #[serde(deserialize_with = "Image::parse_from_string")]
     image: Image,
     env: Option<Vec<String>>,
+    // TODO: rename this field because it does not match to volumes any more (it is file content)
```
Comment (qacwnfq) [inline api/src/models/service.rs]: Are we tracking this TODO anywhere (github issue or JIRA)?

File: api/src/services/config_service.rs [hunk 2]
```diff
@@ -45,6 +52,7 @@ pub struct ContainerConfig {
 pub struct JiraConfig {
     host: String,
     user: String,
+    // TODO: use secstr
```
Comment (x1B) [inline api/src/services/config_service.rs]: I guess that this definitely warrants an issue?

File: api/src/services/config_service.rs [hunk 3]
```diff
@@ -221,14 +298,40 @@ impl TryFrom<&Companion> for ServiceConfig {
...
+        ( $name:expr ) => {{
+            let mut hasher = Sha256::new();
+            hasher.input($name);
+            let img_hash = &format!("sha256:{:x}", hasher.result_reset());
+
+            ServiceConfig::new(String::from($name), Image::from_str(&img_hash).unwrap())
+        }};
+    }
+
+    macro_rules! config_from_str {
```
Comment (x1B) [inline api/src/services/config_service.rs]: I didn't know creating macros would be so simple :+1:

File: api/src/services/config_service.rs [hunk 4]
```diff
@@ -343,22 +446,148 @@ mod tests {
...
     }
+
+    #[test]
+    fn should_return_application_companions_with_specific_app_selector() {
+        let config = config_from_str!(
+            r#"
+            [companions.openid]
+            serviceName = 'openid'
+            type = 'application'
+            image = 'private.example.com/library/opendid:latest'
```
Comment (x1B) [inline api/src/services/config_service.rs]: 🦅 👁 not a big deal, but `opendid`->`openid`?

File: api/src/services/config_service.rs (hunk 3 above)
Comment (schrieveslaach) [reply api/src/services/config_service.rs]: Rust is just awesome. :wink:

File: api/src/models/service.rs (hunk 1 above)
Comment (schrieveslaach) [reply api/src/models/service.rs]: I opened an issue: #8

File: api/src/services/config_service.rs (hunk 2 above)
Comment (schrieveslaach) [reply api/src/services/config_service.rs]: That was a leftover... Now, it is fixed.

File: api/src/services/config_service.rs (hunk 4 above)
Comment (schrieveslaach) [reply api/src/services/config_service.rs]: Nice catch. :+1: :eagle:

**Pull request merged**